testapi.pm

```diff
@@ -1030,7 +1030,9 @@ C<$serialdev>.
 sub assert_script_sudo {    # no:style:signatures
     my ($cmd, $wait) = @_;
     my $str = hashed_string("ASS$cmd");
-    script_sudo("$cmd; echo $str-\$?- > /dev/$serialdev", 0);
+    my $redirect_to_serial_console = if is_serial_terminal() ? '' : "> /dev/$serialdev";
```

Comment:
The `if` should be removed:
```suggestion
    my $redirect_to_serial_console = is_serial_terminal() ? '' : "> /dev/$serialdev";
```

Reply:
slight correction:
```suggestion
    my $redirect_to_serial_console = is_serial_terminal() ? '' : " > /dev/$serialdev";
```
and then what @okurz suggested.

Reply:
*ready for review*